glean/src/core/pings/index.ts

```diff
@@ -60,10 +112,43 @@ class PingType implements CommonPingData {
         console.error(`Invalid reason code ${reason} from ${this.name}. Ignoring.`);
         correctedReason = undefined;
       }
-  
+
       const identifier = generateUUIDv4();
       await collectAndStorePing(identifier, this, correctedReason);
+
+      // This guarantees that, when running tests, the promise returned by
+      // `testBeforeNextSubmit` is resolved after the ping is collected: this is
+      // needed to make sure calling the testing APIs on metrics behave consistently
+      // if tests run fast.
+      if (testResolver) {
+        testResolver();
+      }
+    });
+  }
+
+  /**
+   * **Test-only API**
+   *
+   * Runs a validation function before the ping is collected.
+   *
+   * TODO: Only allow this function to be called on test mode (depends on Bug 1682771).
+   *
+   * @param validatorFn The asynchronous validation function to run in order to validate
+   *                    the ping content.
+   *
+   * @returns A `Promise` resolved when the ping is collected and the validation function
+   *          is executed.
+   */
+  async testBeforeNextSubmit(validatorFn: ValidatorFunction): Promise<void> {
+    if (this.testValidator) {
+      console.error(`There is an existing test call for ping "${this.name}": ignoring.`);
```

Comment:
```suggestion
      console.error(`There is an existing test call for ping "${this.name}". Ignoring.`);
```

Just to keep consistency with other messages around the code.